src/disco/topo/fd_topo_run.c

```diff
@@ -161,22 +161,27 @@ fd_topo_run_tile( fd_topo_t *          topo,
   if( FD_UNLIKELY( tile_run->lazy ) ) lazy = tile_run->lazy( tile_mem );
 
   fd_rng_t rng[1];
-  fd_mux_tile( tile->cnc,
-               tile_run->mux_flags,
-               polled_in_cnt,
-               in_mcache,
-               in_fseq,
-               tile->out_link_id_primary == ULONG_MAX ? NULL : topo->links[ tile->out_link_id_primary ].mcache,
-               out_cnt_reliable,
-               out_fseq,
-               tile_run->burst,
-               0,
-               lazy,
-               fd_rng_join( fd_rng_new( rng, 0, 0UL ) ),
-               fd_alloca( FD_MUX_TILE_SCRATCH_ALIGN, FD_MUX_TILE_SCRATCH_FOOTPRINT( polled_in_cnt, out_cnt_reliable ) ),
-               ctx,
-               &callbacks );
-  FD_LOG_ERR(( "tile run loop returned" ));
+  int ret = 0;
+  if( FD_LIKELY( tile_run->main == NULL ) ) {
+    ret = fd_mux_tile( tile->cnc,
```

Comment:
This is not the right way to do this, and has a few problems. You can see them above: the threads don't get their own metrics, or sandbox, the IDs might be assigned incorrectly and cause deadlocks.

You should spawn actual tiles for the worker threads, but just run them in the same process. You can do this with a new flag on the topo, and using the existing system that run_solana does.

Reply:
fd_topo_run launcher is only used for debugging, as well, so this is broken for the production fdctl boot, which is in `run.c`

Reply:
You can't actually spawn the tile thread this early. You need to wait until tpool is created.